app/views/__init__.py

```diff
@@ -0,0 +1,172 @@
+    result= state.search_states(name)
+    if result:
+        return jsonify({"state": result, "code": 200})
+    else:
+        return jsonify({"message": "state not found", "code": 404})
+        
+@state_bp.route("/api/v1/states/capital", methods=["GET"])
+def states_capital():
+    state = State()
+    capitals = state.list_state_capitals()
+    if capitals:
+        return jsonify({"message": "capitals displayed successfully", "status": 201,"result": capitals})
+    else:
+        return jsonify({"message": "capitals not displayed", "status": 400})
+
+@state_bp.route("/api/v1/states/average_population", methods=["GET"])
+def average_states_population():
+    """
+    this function calculate and return the average population of states
+    """
+    state = State()
+    average_pop = state.average_population()
+    if average_pop:
+        return jsonify({"message": "average population displayed successfully", "status": 201,"result": average_pop})
+    else:
+        return jsonify({"message": "average population not displayed", "status": 400})
+
+@state_bp.route("/api/v1/states/range_population", methods=["GET"])
+def range_of_population():
+    """
+    that counts the number of states within 
+a population between 1,000,000 and 5,000,000 people
+    """
+    state = State()
+    range_count = state.count_population_range()
+    if range_count:
+        return jsonify({"message": "range of population displayed successfully", "status": 201,"result": range_count})
+    else:
+        return jsonify({"message": "range of population not displayed", "status": 400})
+
+@state_bp.route("/api/v1/states/join_state_capital", methods=["GET"])
+def joinStateCapital():
+    join_record = State().join_states_capital()
+    if join_record:
+        return jsonify({"message": "joined records displayed successfully", "status": 201,"result": join_record})
+    else:
+        return jsonify({"message": "joined records not displayed", "status": 400})
+   
+
+@state_bp.route("/api/v1/states/searchByName",methods=["POST"])
+def searchStateByName():
+    """
+    this functions searches for state details by name
+    """
+    state=State()
+
+    data=request.get_json()
+
+    name=data.get("name")
+
+    result=state.search_State_By_Name(name)
+
+    if result:
+        return jsonify({"message": "search successful", "status": 201,"result":result})
+    else:
+        return jsonify({"message": "search unsuccessful", "status": 400,"result":result})
+    
+@state_bp.route("/api/v1/states/admitedBtw1750and1850",methods=["GET"])
+def stateAdmittedBetween1750and1850():  
+    """
+    this functions searches for state details admitted between 1750 and 1850
+    """
+    state=State() 
+    result = state.state_admitted_btw_1750_AND_1850()
+    if result:
+        return jsonify({"message": "search successful", "status": 201,"result":result})
+    else:
+        return jsonify({"message": "search unsuccessful", "status": 400,"result":result})
+    
```

Comment:
- Use `make_response` function to format your server response
- Shorten your API urls in your view using `url_prefix` where you are registering your blueprint

Reply:
Working on changes